src/Orckestra.Composer.Search/Providers/FromPriceProvider.cs

```diff
@@ -168,7 +175,13 @@ private static string GetPriceListId(ProductDocument document)
 
         private string GetDisplayPrice(double? price)
         {
-            return !price.HasValue ? null : price.Value.ToString("C2", ComposerContext.CultureInfo);
+            if (price.HasValue)
+            {
+                decimal.TryParse(price.Value.ToString(CultureInfo.InvariantCulture), out var priceVal);
```

Comment:
That's a rather unusual way of converting a double value into a decimal one. You can probably just use the cast operator:

`return LocalizationProvider.FormatPrice((decimal) price.Value, CurrencyProvider.GetCurrency());`